### DISCUSSION TITLE
[ASK] Suggestion - how to further format and stylize generated XLSX file?

### DISCUSSION THREAD

**Discussion by andrej-gravik** [pxlrbt/filament-excel#149] [Q&A] [answered]
I guess colouring the excell cells and aligning them is beyond the scope of this library but if I want to do so, what do you suggest?

1. Where should I hook to edit generated XLSX file before downloading it?
2. What library should I use for editing / formating the excel file. Is there existing dependency of this module I could use or do I have to install totally another library for it?

Thanks for any suggestions!

**Comment by pxlrbt** [comment]
Not sure whether it's the ideal way of doing so, but I think you could hook into the generation via the `BeforeSheet` event. Somebody posted something similar on Discord.

```php
Sheet::listen(BeforeSheet::class, function ($sheet) {
    $sheet->getDelegate()->getStyle('1')->getFont()->setSize(8);
});
```

Or you extend the Export class and add the required methods to it and use this class.

**Reply by andrej-gravik** [reply]
Thanks, It helped a lot :)

**Reply by pxlrbt** [reply]
If you find a solution that works for you, feel free to share it here.

**Reply by andrej-gravik** [reply, marked as answer]
Basicly What you suggested and then add some sugar around it to make it work :)

```php

// Implementing WithEvents interface is necessary
class WorkPresenceReportExport extends ExcelExport implements WithEvents
{
    // do not forget use RegistersEventListeners trait
    use RegistersEventListeners;

    // ... REST OF MY CUSTOM EXPORT CODE ...

    public function registerEvents(): array
    {
        return [
            AfterSheet::class => function (AfterSheet $excel) {
                // Get sheet object
                $sheet = $excel->getSheet();

                // Insert rows
                $sheet->insertNewRowBefore(1, 4);

                // Align cell text to center
                $sheet->getStyle('C4')->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);

                // Column background color
                $sheet->getStyle('D')->getFill()->applyFromArray([
                    'fillType' => 'solid',
                    'color' => ['rgb' => 'FF00AA']
                 ]);

                // ETC...
            }
        ];
    }
}
